2.42.0 Release Branch: Flink Examples integration test failing

### What happened?

Release Validation branch https://github.com/apache/beam/pull/23200 has failures for the Flink Example post commits:

https://ci-beam.apache.org/job/beam_PostCommit_Java_Examples_Flink_PR/8/

from the beam branch: `./gradlew :runners:flink:1.13:examplesIntegrationTest` replicates the same failure.

Errors are all the following InvalidProgramException

```
08:52:14 org.apache.beam.examples.snippets.transforms.io.gcp.bigquery.BigQuerySamplesIT > testTableIO FAILED
08:52:14     org.apache.beam.sdk.Pipeline$PipelineExecutionException at BigQuerySamplesIT.java:108
08:52:14         Caused by: org.apache.flink.api.common.InvalidProgramException at BigQuerySamplesIT.java:108
```

Other flink validation is passing so it appears related to gradle configuration or similar. This needs to be determined and or resolved before RC2 is built.

### Issue Priority

Priority: 1

### Issue Component

Component: runner-flink

--- a/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java
+++ b/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java
@@ -188,7 +188,8 @@ class BatchLoads<DestinationT, ElementT>
     this.rowWriterFactory = rowWriterFactory;
     schemaUpdateOptions = Collections.emptySet();
     this.tempDataset = tempDataset;
-    this.tableDestinationCoder = clusteringEnabled ? TableDestinationCoderV3.of(): TableDestinationCoderV2.of();
+    this.tableDestinationCoder =
+        clusteringEnabled ? TableDestinationCoderV3.of() : TableDestinationCoderV2.of();
   }
 
   void setSchemaUpdateOptions(Set<SchemaUpdateOption> schemaUpdateOptions) {

--- a/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java
+++ b/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java
@@ -146,8 +146,8 @@ class BatchLoads<DestinationT, ElementT>
   private final Coder<ElementT> elementCoder;
   private final RowWriterFactory<ElementT, DestinationT> rowWriterFactory;
   private final @Nullable String kmsKey;
-  private final boolean clusteringEnabled;
   private final String tempDataset;
+  private Coder<TableDestination> tableDestinationCoder;
 
   // The maximum number of times to retry failed load or copy jobs.
   private int maxRetryJobs = DEFAULT_MAX_RETRY_JOBS;
@@ -186,9 +186,9 @@ class BatchLoads<DestinationT, ElementT>
     this.elementCoder = elementCoder;
     this.kmsKey = kmsKey;
     this.rowWriterFactory = rowWriterFactory;
-    this.clusteringEnabled = clusteringEnabled;
     schemaUpdateOptions = Collections.emptySet();
     this.tempDataset = tempDataset;
+    this.tableDestinationCoder = clusteringEnabled ? TableDestinationCoderV3.of(): TableDestinationCoderV2.of();
   }
 
   void setSchemaUpdateOptions(Set<SchemaUpdateOption> schemaUpdateOptions) {
@@ -493,7 +493,8 @@ class BatchLoads<DestinationT, ElementT>
                             maxRetryJobs,
                             kmsKey,
                             loadJobProjectId))
-                    .withSideInputs(copyJobIdPrefixView));
+                    .withSideInputs(copyJobIdPrefixView))
+            .setCoder(tableDestinationCoder);
 
     PCollectionList<TableDestination> allSuccessfulWrites =
         PCollectionList.of(successfulSinglePartitionWrites).and(successfulMultiPartitionWrites);
@@ -754,9 +755,6 @@ class BatchLoads<DestinationT, ElementT>
       PCollectionView<String> loadJobIdPrefixView) {
     List<PCollectionView<?>> sideInputs = Lists.newArrayList(loadJobIdPrefixView);
     sideInputs.addAll(dynamicDestinations.getSideInputs());
-
-    Coder<TableDestination> tableDestinationCoder =
-        clusteringEnabled ? TableDestinationCoderV3.of() : TableDestinationCoderV2.of();
 
     Coder<KV<ShardedKey<DestinationT>, WritePartition.Result>> partitionsCoder =
         KvCoder.of(

